hypershift-operator/controllers/machineconfigserver/machineconfigserver_controller.go

```diff
@@ -227,29 +226,10 @@ func (r *MachineConfigServerReconciler) Reconcile(ctx context.Context, req ctrl.
 	return ctrl.Result{}, nil
 }
 
-func (r *MachineConfigServerReconciler) reconcileMCSServiceNodePort(ctx context.Context, mcs *hyperv1.MachineConfigServer, mcsService *corev1.Service) (*ctrl.Result, error) {
+func (r *MachineConfigServerReconciler) reconcileMCSServiceNodePortResources(ctx context.Context, mcs *hyperv1.MachineConfigServer, mcsService *corev1.Service) (*ctrl.Result, error) {
 	r.Log.Info("Creating MCS Service")
 	_, err := controllerutil.CreateOrPatch(ctx, r.Client, mcsService, func() error {
-		mcsService.Spec.Ports = []corev1.ServicePort{
-			{
-				Name:       "http",
-				Protocol:   corev1.ProtocolTCP,
-				Port:       80,
-				TargetPort: intstr.FromInt(8080),
-			},
-		}
-		mcsService.Spec.Selector = map[string]string{
-			"app": fmt.Sprintf("machine-config-server-%s", mcs.Name),
-		}
-		mcsService.Spec.Type = corev1.ServiceTypeNodePort
-
-		// If there's user input nodePort and there's no existing one, we set it.
-		if (mcs.Spec.IgnitionService.NodePort != nil && mcs.Spec.IgnitionService.NodePort.Port > 0) &&
-			mcsService.Spec.Ports[0].NodePort <= 0 {
-			mcsService.Spec.Ports[0].NodePort = mcs.Spec.IgnitionService.NodePort.Port
-		}
-
-		return nil
+		return r.reconcileMCSServiceNodePort(mcsService, mcs)
```

Comment:
Agree, we should only modify the spec in the reconcile function

Reply:
Ok that should be done as is in this design: What I can do is `reconcileResouces` naming might be confusing: The actual reconciling is done in the lower level functions
`reconcileKubeAPIServerServiceLoadBalancer, reconcileVPNServerServiceNodePort,` and the higher level `Resources` function is where after reconciliation the data is gathered and inserted into the appropriate status, etc.

I can try and rename them to clarify this piece but not sure of a good name. I could do `gatherMetadata` instead of reconcile. Would that work? 
So 
 ensureInfrastructure -> gatherMetadataKubeAPIServerServiceNodePortResources -> reconcileKubeAPIServerServiceLoadBalancer